server/container/mailbox-jmx/src/test/java/org/apache/james/adapter/mailbox/ReIndexerManagementTest.java

```diff
@@ -0,0 +1,58 @@
+package org.apache.james.adapter.mailbox;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.indexer.ReIndexer;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.task.Task;
+import org.apache.james.task.TaskId;
+import org.apache.james.task.TaskManager;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ReIndexerManagementTest {
+    private ReIndexerManagement testee;
+    private TaskManager taskManager;
+    private ReIndexer reIndexer;
+    
+    @Before
+    public void setUp() {
+        taskManager = mock(TaskManager.class);
+        reIndexer = mock(ReIndexer.class);
```

Comment:
I looked the `ReIndexerImplTest`, I think you could use this setup to avoid mock when possible
```
    @BeforeEach
    void setUp() throws MailboxException {
        mailboxManager = new InMemoryIntegrationResources().createMailboxManager(new SimpleGroupMembershipResolver());
        MailboxSessionMapperFactory mailboxSessionMapperFactory = mailboxManager.getMapperFactory();
        messageSearchIndex = mock(ListeningMessageSearchIndex.class);
        reIndexer = new ReIndexerImpl(new ReIndexerPerformer(mailboxManager, messageSearchIndex, mailboxSessionMapperFactory));
    }
```

Reply:
sorry, what is the problem when you're still using mock instead of some real objects? like `ReIndexer` and `Task`?

Reply:
To avoid side effects (executing something, etc.)